### PR TITLE
Add details about 'atleast as good as f32' for AbstractFloats

### LINKED ISSUE #3111: Add discussion of AbstractFloat behaviour to `fp_primer.md`
For ULP and absolute errors, AF behave like f32. What this means and the implications should be discussed in the primer.

### PR DESCRIPTION
**Pull request by zoddicus**
This also removes a dead link

Fixes #3111

<hr>

**Requirements for PR author:**

- [x] All missing test coverage is tracked with "TODO" or `.unimplemented()`.
- [x] New helpers are `/** documented */` and new helper files are found in `helper_index.txt`.
- [x] Test behaves as expected in a WebGPU implementation. (If not passing, explain above.)

**Requirements for [reviewer sign-off](https://github.com/gpuweb/cts/blob/main/docs/reviews.md):**

- [ ] Tests are properly located in the test tree.
- [ ] [Test descriptions](https://github.com/gpuweb/cts/blob/main/docs/intro/plans.md) allow a reader to "read only the test plans and evaluate coverage completeness", and accurately reflect the test code.
- [ ] Tests provide complete coverage (including validation control cases). **Missing coverage MUST be covered by TODOs.**
- [ ] Helpers and types promote readability and maintainability.

When landing this PR, be sure to make any necessary issue status updates.

### CODE REVIEW

File: docs/fp_primer.md [hunk 1]
```diff
@@ -690,6 +690,42 @@ library, or if this turns out to be a significant issue in the future, this
...
+AbstractFloat internally is defined as a f64, and this applies for exact and
+correctly rounded accuracies. Thus, correctly rounded refers to the nearest f64
+values. Where AbstractFloat is different is for ULP and absolute errors. Reading
+the spec strictly, these all have unbounded accuracies, but is recommended that
+their accuracies be at least as good as the f32 equivalent.
+
+The difference between f32 and f64 ULP at a specific value X are significant, so
+at least as good as f32 requirement is always less strict than if it was
+calculated in terms of f64. Similarly, for absolute accuracies the interval
+`[x - episilon, x + episilon]` is always equal or greater if calculated as f32s
```
Comment (kainino0x) [inline docs/fp_primer.md:710]:
```suggestion
`[x - epsilon, x + epsilon]` is always equal or wider if calculated as f32s
```

File: docs/fp_primer.md [hunk 2]
```diff
@@ -690,6 +690,42 @@ library, or if this turns out to be a significant issue in the future, this
...
+### Accuracy
+
+For the concrete floating point types (f32 & f16) the accuracy of operations are
+defined in terms of their own type. Specifically for f32, correctly rounded
+refers to the nearest f32 values, and ULP is in terms of the distance between
+f32 values.
+
+AbstractFloat internally is defined as a f64, and this applies for exact and
+correctly rounded accuracies. Thus, correctly rounded refers to the nearest f64
+values. Where AbstractFloat is different is for ULP and absolute errors. Reading
```
Comment (kainino0x) [inline docs/fp_primer.md:703]:
```suggestion
values. However, AbstractFloat differs for ULP and absolute errors. Reading
```

File: docs/fp_primer.md [hunk 3]
```diff
@@ -690,6 +690,42 @@ library, or if this turns out to be a significant issue in the future, this
...
+`[x - episilon, x + episilon]` is always equal or greater if calculated as f32s
+vs f64s.
+
+If an inherited accuracy is only defined in terms of correctly rounded
+accuracies, then the interval is calculated in terms of f64s. If any of the
+defining accuracies are ULP or absolute errors, then the result falls into the
+unbounded accuracy, but recommended to be at least as good as f32 bucket.
+
+What this means from a CTS implementation is that for these at least as good as
+f32 error intervals, if the infinitely accurate result is finite for f32, then
```
Comment (kainino0x) [inline docs/fp_primer.md:719]:
```suggestion
What this means from a CTS implementation is that for these "at least as good as
f32" error intervals, if the infinitely accurate result is finite for f32, then
```

**Review by kainino0x — COMMENTED** [review]
I'm not that familiar with either FPTraits or the WGSL spec here, but otherwise LGTM

File: docs/fp_primer.md [hunk 4]
```diff
@@ -690,6 +690,42 @@ library, or if this turns out to be a significant issue in the future, this
...
+
+For the concrete floating point types (f32 & f16) the accuracy of operations are
+defined in terms of their own type. Specifically for f32, correctly rounded
+refers to the nearest f32 values, and ULP is in terms of the distance between
+f32 values.
+
+AbstractFloat internally is defined as a f64, and this applies for exact and
+correctly rounded accuracies. Thus, correctly rounded refers to the nearest f64
+values. However, AbstractFloat differs for ULP and absolute errors. Reading
+the spec strictly, these all have unbounded accuracies, but is recommended that
```
Comment (dneto0) [inline docs/fp_primer.md:704]: nit:  should be " but it is recommended that "  (insert the "it")

**Review by dneto0 — APPROVED** [review]
This is explained very well. Thank you!

File: docs/fp_primer.md (hunk 4 above)
Comment (dneto0) [reply docs/fp_primer.md:704]: alternately, "but it recommends that..." (where "it" refers to the spec)

**Pull request merged**